components/support/ffi/src/lib.rs

```diff
@@ -343,3 +343,81 @@ fn init_backtraces_once() {
 
 #[cfg(not(feature = "log_backtraces"))]
 fn init_backtraces_once() {}
+
+/// ByteBuffer is a struct that represents an array of bytes to be sent over the FFI boundaries.
+///
```

Comment:
Nit: Use case description. Something like

```suggestion
///
/// There are several cases when you might want to use this, but the primary one for us
/// is for returning protobuf-encoded data to Swift and Java. The type is currently rather
/// limited (implementing almost no functionality), however in the future it may be
/// more expanded.
```